### PR TITLE
Video listing page

### PR DESCRIPTION
**Pull request by AbhishekkGautam**
### Video Listing Page
- JSON data 
`id, title, channelName, views, uploadedOn, categoryName, avatar, description.`
- Navbar - `logo, search-input, profile-icon, login button`
- Sidebar 
  - Home
  - Trending
  - History
  - Liked videos
  - Watch later
  - Playlist
 - Category tags ( only css )
 - Video listing component
 - Video card
   - thumbnail
   - avatar ( channel profile pic )
   - title
   - views count ( custom formatted like - `40K` , `3.2M`  )
   - date ( formatted like - `a month ago`, `3 years ago` )
   - `more-options` icon for modal with `save to playlist`, `save to watch later` & `share` options.
   
### Desktop - 
mobile (4)

### Mobile screen with modal - 
mobile (5)

### CODE REVIEW

File: src/context/VideoContext.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+  useEffect(() => {
+    (async () => {
+      try {
+        const { data, status } = await axios.get("api/videos");
+        if (status === 200) {
+          dispatch({ type: "LOAD_ALL_VIDEOS", payload: data.videos });
+        }
+      } catch (error) {
+        dispatch({ type: "ERROR", payload: error });
```
Comment (gopalbbharadva) [inline src/context/VideoContext.js:24]: Suggestion: You have added variables for type values but might be forgot to add it here.

File: src/components/VideoListing/MoreOptionsModal.js [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+        setShowOptionsModal(false);
+      }
+    };
+    document.addEventListener("mousedown", checkIfClickedOutside);
+
+    return () => {
+      // Cleanup the event listener
+      document.removeEventListener("mousedown", checkIfClickedOutside);
+    };
+  }, [showOptionsModal]);
```
Comment (gopalbbharadva) [inline src/components/VideoListing/MoreOptionsModal.js:21]: Question: Is it the only way to toggle model as it is in vanilla js in React. we should avoid vanilla js in react right?

**Review by gopalbbharadva — APPROVED** [review]
All Files and code is well - structured. 1 number.

File: src/components/VideoListing/MoreOptionsModal.js (hunk 2 above)
Comment (AbhishekkGautam) [reply src/components/VideoListing/MoreOptionsModal.js:21]: I've tried to do it with only react but couldn't get the desired result. 
I want the modal to close whenever user clicks anywhere outside the modal. 
And many blogs have suggested this way. 
JSConf - https://www.youtube.com/watch?v=J-g9ZJha8FE&feature=youtu.be&t=481 
So we might have to use some vanilla js for specific use cases.

File: src/context/VideoContext.js (hunk 1 above)
Comment (AbhishekkGautam) [reply src/context/VideoContext.js:24]: Will refactor it.

File: public/manifest.json [hunk 3]
```diff
@@ -0,0 +1,25 @@
+{
+  "short_name": "React App",
+  "name": "Create React App Sample",
```
Comment (msdeshmukh009) [inline public/manifest.json:3]: You can put the details of your project over here.

**Review by msdeshmukh009 — APPROVED** [review]
Everything looks fine to me. 
The code looks readable and clean.
Good to merge.

File: public/manifest.json (hunk 3 above)
Comment (AbhishekkGautam) [reply public/manifest.json:3]: Yes. I'll do it. I need to add favicon as well.

**Pull request merged**